src/Controllers/UserProgressController.php

```diff
@@ -0,0 +1,298 @@
+use Railroad\Railtracker\Trackers\MediaPlaybackTracker;
+
+class UserProgressController extends Controller
+{
+
+    /**
+     * @var ContentService
+     */
+    private $contentService;
+
+    /**
+     * @var UserContentProgressService
+     */
+    private $userContentProgressService;
+
+    /**
+     * @var MediaPlaybackTracker
+     */
+    private $mediaPlaybackTracker;
+
+    /**
+     * UserProgressController constructor.
+     *
+     * @param ContentService $contentService
+     * @param UserContentProgressService $userContentProgressService
+     * @param MediaPlaybackTracker $mediaPlaybackTracker
+     */
+    public function __construct(
+        ContentService $contentService,
+        UserContentProgressService $userContentProgressService,
+        MediaPlaybackTracker $mediaPlaybackTracker
+    ) {
+        $this->contentService = $contentService;
+        $this->userContentProgressService = $userContentProgressService;
+        $this->mediaPlaybackTracker = $mediaPlaybackTracker;
+    }
+
+    /**
+     * @param Request $request
+     * @return JsonResponse
+     * @throws NonUniqueResultException
+     */
+    public function completeUserProgressOnContent(Request $request)
+    {
+        ModeDecoratorBase::$decorationMode = DecoratorInterface::DECORATION_MODE_MINIMUM;
+
+        $rules = [
+            'content_id' => 'required',
+            'device_type' => 'required',
+        ];
+
+        $validator = Validator::make(
+            $request->all(),
+            $rules
+        );
+
+        if ($validator->fails()) {
+            return response()->json(
+                [
+                    'success' => false,
+                    'title' => 'Complete Failed',
+                    'message' => implode(
+                        ',',
+                        $validator->getMessageBag()
+                            ->all()
+                    ),
+                ],
+                422
+            );
+        }
+
+        $content = $this->contentService->getById($request->get('content_id'));
+
+        if (empty($content)) {
+            return response()->json($content);
+        }
+
+
+        $this->userContentProgressService->completeContent(
+            $request->get('content_id'),
+            auth()->id()
+        );
+
+        if ($content['type'] != 'assignment') {
+            ModeDecoratorBase::$decorationMode = DecoratorInterface::DECORATION_MODE_MAXIMUM;
+
+            $parentType = $this->getLessonTypeParentType($content['type']);
+
+            $parent = $this->contentService->getByChildIdWhereType($request->get('content_id'), $parentType);
+        } else {
+            $parent = $this->contentService->getByChildId($request->get('content_id'));
+        }
+
+        $response =  [
+            'success' => true,
+            'parent' => $parent,
+        ];
+
+        if(config('musora-api.shouldDisplayReview', false)) {
+            $user = $this->userRepository->findOneBy(['id' => auth()->id()]);
```

Comment:
Some of these class attributes do not exist. I am not sure this controller will function.